lib/internal/module.js

```diff
@@ -44,3 +48,31 @@ function stripBOM(content) {
   }
   return content;
 }
+
+exports.builtinLibs = ['assert', 'buffer', 'child_process', 'cluster',
+  'crypto', 'dgram', 'dns', 'domain', 'events', 'fs', 'http', 'https', 'net',
+  'os', 'path', 'punycode', 'querystring', 'readline', 'repl', 'stream',
+  'string_decoder', 'tls', 'tty', 'url', 'util', 'v8', 'vm', 'zlib'];
+
+function addBuiltinLibsToObject(object) {
+  // Make built-in modules available directly (loaded lazily).
+  exports.builtinLibs.forEach((name) => {
+    Object.defineProperty(object, name, {
+      get: () => {
+        const lib = require(name);
+        // This implicitly invokes the setter, so that this getter is only
+        // invoked at most once and does not overwrite anything.
+        object[name] = lib;
+        return lib;
+      },
+      // Allow the creation of other globals with this name.
+      set: (val) => {
+        // Deleting the property before re-assigning it disables the
+        // getter/setter mechanism.
+        delete object[name];
+        object[name] = val;
+      },
+      configurable: true
```

Comment:
> Perhaps just make sure it stays non-enumerable.

@bnoordhuis That’s probably doable, but I think it would get a lot trickier to maintain the current behaviour of allowing to re-assign these globals. Correct me if I’m wrong, but to do this, you’d have to:
* Stop invoking the setter from the getter, and
* Call `delete` from there “manually”, and use `Object.defineProperty({ value: … })` to set the reference to the module, and
* Give that new property descriptor the setter that is used right now, too, so that in the case of user-defined globals with these names they *are* enumerable as expected

Maybe there’s a simpler way, and I agree that it’s a bit weird the way it is right now (and has been for the last three years), but I don’t think anybody really cares very deeply.